examples/stress/runs_per_second.py

```diff
@@ -0,0 +1,100 @@
+import asyncio
+import time
+
+from aiolimiter import AsyncLimiter
+
+import flyte
+
+env = flyte.TaskEnvironment(
+    name="runs_per_second",
+    resources=flyte.Resources(cpu=1, memory="1Gi"),
+)
+
+downstream_env = flyte.TaskEnvironment(
+    name="downstream",
+    resources=flyte.Resources(cpu=1, memory="1Gi"),
+    reusable=flyte.ReusePolicy(
+        replicas=10,
+        idle_ttl=60,
+        concurrency=10,
+        scaledown_ttl=60,
+    ),
+    image=flyte.Image.from_debian_base(install_flyte=False).with_pip_packages(
+        "unionai-reuse==0.1.6b0", "flyte>2.0.0b21", pre=True
+    ),
+)
+
+
+@downstream_env.task
+async def sleeper(x: int) -> int:
+    await asyncio.sleep(1.0)
+    return x
+
+
+@env.task
+async def runs_per_second(max_rps: int = 50, n: int = 500):
+    """
+    Measure actual runs per second while throttling above max_rps.
+    Prints statistics to calibrate sequential run creation performance.
+    """
+    limiter = AsyncLimiter(max_rps, 1)
+
+    start_time = time.time()
+    run_times = []
+
+    print(f"Starting {n} runs with max RPS throttling at {max_rps}")
+    print("=" * 60)
+
+    for i in range(n):
+        run_start = time.time()
+
+        async with limiter:
+            # Create the run (this is the operation we're measuring)
+            await flyte.run.aio(sleeper, x=i)
+
+        run_end = time.time()
+        run_times.append(run_end - run_start)
+
+        # Print progress every 50 runs
+        if (i + 1) % 50 == 0:
+            elapsed = run_end - start_time
+            current_rps = (i + 1) / elapsed
+            avg_run_time = sum(run_times[-50:]) / min(50, len(run_times))
+            print(f"Runs {i + 1:3d}: {current_rps:6.2f} RPS | Avg run time: {avg_run_time * 1000:6.2f}ms")
```

Comment:
calls for a report to be produced with a beautiful graph :D

Reply:
Aah let me do that